src/Avalonia.Controls/ProgressBar.cs

```diff
@@ -29,16 +63,80 @@ protected override Size ArrangeOverride(Size finalSize)
         protected override void OnTemplateApplied(TemplateAppliedEventArgs e)
         {
             _indicator = e.NameScope.Get<Border>("PART_Indicator");
+
             UpdateIndicator(Bounds.Size);
+            UpdateOrientation(Orientation);
+            UpdateIsIndeterminate(IsIndeterminate);
         }
 
         private void UpdateIndicator(Size bounds)
         {
             if (_indicator != null)
             {
-                double percent = Maximum == Minimum ? 1.0 : (Value - Minimum) / (Maximum - Minimum);
-                _indicator.Width = bounds.Width * percent;
+                if (IsIndeterminate)
+                {
+                    if (Orientation == Orientation.Horizontal)
+                        _indicator.Width = bounds.Width / 5.0;
+                    else
+                        _indicator.Height = bounds.Height / 5.0;
+                }
+                else
+                {
+                    double percent = Maximum == Minimum ? 1.0 : (Value - Minimum) / (Maximum - Minimum);
+
+                    if (Orientation == Orientation.Horizontal)
+                        _indicator.Width = bounds.Width * percent;
+                    else
+                        _indicator.Height = bounds.Height * percent;
+                }
+            }
+        }
+
+        private void UpdateOrientation(Orientation orientation)
+        {
+            if (orientation == Orientation.Horizontal)
+            {
+                MinHeight = 14;
+                MinWidth = 200;
+
+                _indicator.HorizontalAlignment = HorizontalAlignment.Left;
+                _indicator.VerticalAlignment = VerticalAlignment.Stretch;
+            }
+            else
+            {
+                MinHeight = 200;
+                MinWidth = 14;
+
+                _indicator.HorizontalAlignment = HorizontalAlignment.Stretch;
+                _indicator.VerticalAlignment = VerticalAlignment.Bottom;
+            }
+        }
+
+        private void UpdateIsIndeterminate(bool isIndeterminate)
+        {
+            if (isIndeterminate)
```

Comment:
I'm afraid, that would lead to a memory leak: if you have an `ProgressBar` with `IsIndeterminate = true` and close the window or simply remove it from the visual tree, the animation will keep ticking and holding a reference to `_indicator`. Which, in turn, has a reference to its parent and so on.

Consider using `WeakReference<T>` for holding an reference to the `ProgressBar` and destroying  subscription when it's gone.

Reply:
I don't know how to use `WeakReference<T>` in this case. Can you help me, please?

Reply:
You need to move `UpdateIsIndeterminate` to a nested class that has a field of `WeakReference<ProgressBar>` type and also move `_indeterminateBindSubscription` field there.

Then animation fires and `WeakReference.Target == null`, you need to automatically dispose it.